src/compartment.c

```diff
@@ -55,6 +55,13 @@ entry_point_cmp(const void* val1, const void* val2)
 struct Compartment*
 comp_from_elf(char* filename, struct ConfigEntryPoint* entry_points, size_t entry_point_count)
 {
+    // Check file exists
+    if (access(filename, F_OK) != 0)
```

Comment:
This is probably Ok for a demo, but it's normally better to try the `open()`, and handle errors there, and that seems an easy change to make. The problem is that the permissions could change between `access()` and `open()`, so in some contexts this pattern can be a security issue.

Reply:
There's a ton of discussion on this topic, since I also looked up what the best way would be [1]. Honestly, I don't have too much strong feeling about it - I think I added this just to have a more user-friendly error message, not for any too important reason. There is an `open` further down [2] - I can just move this code there?

[1] https://stackoverflow.com/questions/230062/whats-the-best-way-to-check-if-a-file-exists-in-c
[2] https://github.com/capablevms/CHERI-ELF-comp/pull/12/files/3419b3e08d4c3d108e9c7c09e88af761ce65ad7c#diff-7887211152112561f3aeba0bbee694021f8d663651ae72fe0164649b2ba2ea32L67

Reply:
I think if you just want to know if it exists, `access()` is good; the problems turn up when you _also_ want to `open()` it. Since `open(..., O_RDONLY)` will fail anyway if the file doesn't exist, you might as well just do that:

```
int fd = open(filename, RD_ONLY);
if (fd < 0) {
    printf("File %s does not exist!\n", filename);
    exit(1);
}
```

You probably want to do that before the `malloc` and `comp_init`, though, otherwise they'll leak.

Reply:
Done in https://github.com/capablevms/CHERI-ELF-comp/pull/12/commits/99eb126e95b99e24e872c17de2922c34e0f13eb7. Thanks for spotting this.